### PR TITLE
FIX: use "krogh" as interpolator method-string instead of "krog"

### PR DESCRIPTION
**Pull request by kmuehlbauer**
While working on #8146 this came up. Aligning with upstream naming convention.

- [ ] Closes #xxxx
- [ ] Tests added
- [ ] User visible changes (including notable bug fixes) are documented in `whats-new.rst`
- [ ] New functions/methods are listed in `api.rst`

### CODE REVIEW

File: xarray/tests/test_missing.py [hunk 1]
```diff
@@ -117,7 +117,7 @@ def test_interpolate_pd_compat():
 
 
 @requires_scipy
-@pytest.mark.parametrize("method", ["barycentric", "krog", "pchip", "spline", "akima"])
+@pytest.mark.parametrize("method", ["barycentric", "krogh", "pchip", "spline", "akima"])
 def test_scipy_methods_function(method):
```
Comment (Illviljan) [inline xarray/tests/test_missing.py:121]:
```suggestion
def test_scipy_methods_function(method) -> None:
```

Comment (kmuehlbauer) [reply xarray/tests/test_missing.py]: @Illviljan Thanks for reminding me to check for missing typing. How about the `method` argument. Does it need typing too?

Comment (Illviljan) [reply xarray/tests/test_missing.py]: You could import InterpolantOptions if you want. Does mypy complain if you add a misspelled method without any typing?

File: xarray/core/missing.py [hunk 2]
```diff
@@ -501,7 +501,7 @@ def _get_interpolator(
             )
         elif method == "barycentric":
             interp_class = _import_interpolant("BarycentricInterpolator", method)
-        elif method == "krog":
+        elif method == "krogh":
```
Comment (dcherian) [inline xarray/core/missing.py:504]:
```suggestion
        elif method in ["krogh", "krog"]:
```

**Review by max-sixty — APPROVED** [review]

**Pull request merged**